share/spice/goodreads/goodreads.js

```diff
@@ -0,0 +1,83 @@
+(function(env) {
+  "use strict";
+
+  env.ddg_spice_goodreads = function(api_result) {
+    console.info(api_result);
+
+    // Validating response
+    if(!api_result || api_result.GoodreadsResponse.search["total-results"].text == "0") {
+      return Spice.failed('goodreads');
+    }
+        
+    var get_details_uri = '/js/spice/goodreads/get_book_details/',
+      get_image_for_book_uri = '/js/spice/goodreads/get_image_for_book/';
+
+    $.ajaxSetup({ cache: true });
+    
+    // Render the response
+    Spice.add({
+      id: "goodreads",
+      name: "Books",
+      data: api_result.GoodreadsResponse.search.results.work,
+      meta: {
+        primaryText: "Book results for " + api_result.GoodreadsResponse.search.query.text,
+        sourceName: "GoodReads",
+        sourceUrl: "https://www.goodreads.com/"
+      },
+      
+      
+      onItemShown: function(item) {  
+        // Good reads description API
+        var book = item.best_book;
+  
+        // *** This ajax call is not working ***
+        $.getJSON('/js/spice/goodreads/get_book_details/' + book.id.text, function(data) {
```

Comment:
@iammrigank this works:

```js
            onItemShown: function (item) {

                var book = item.best_book;

                $.getJSON(book_details_uri + book.id.text, function (data) {

                    var description = data.GoodreadsResponse.book.description.text || "-- No Description Found --";

                    item.set({
                        abstract: description && Handlebars.helpers.ellipsis(description.replace(/<\/?\w*>/gm, ''), 400),
                        heading: book.title.text,
                        isbn: data.GoodreadsResponse.book.isbn && data.GoodreadsResponse.book.isbn.text,
                        isbn13: data.GoodreadsResponse.book.isbn13 && data.GoodreadsResponse.book.isbn13.text
                    });

                    $.getJSON(book_image_uri + data.GoodreadsResponse.book.isbn13.text, function (image_object) {
                        item.set({
                            img_m: image_object.results[0].img_m
                        });
                    });
                });
            },
```

Full JS and Perl here: https://gist.github.com/moollaza/1cef2780617d39f7da8cbe817fefe88b